lib/cylc/gui/gscan.py

```diff
@@ -826,8 +826,7 @@ def _get_warnings(self, key):
         failed_tasks = []
         for state in (TASK_STATUS_FAILED, TASK_STATUS_SUBMIT_FAILED):
             failed_tasks += self.tasks_by_state[key].get(state, [])
-        warnings = [warn for warn in failed_tasks if warn[0] > warn_time]
-        warnings.sort()
+        warnings = sorted(warn for warn in failed_tasks if warn[0] > warn_time)
```

Comment:
That's disappointing! It would seem that Python's sorting algorithm needs to analyse the full list before it can begin https://en.wikipedia.org/wiki/Timsort.